### PR TITLE
[Enhancement] Enable skew join rewrite for NULL-only skew

### PR DESCRIPTION
**Pull request by murphyatwork**
## What I'm doing:

Extend stats-based skew join optimization to trigger when the join column’s NULL fraction exceeds the threshold, even if MCVs are empty.
- Consider nullsFraction in skew detection (uses skewJoinDataSkewThreshold)
- Avoid mutating incoming skewValues; add IN branch only when non-empty
- Ensure right-side salt table builds when skewValues is empty/NULL-only

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 4.1
  - [ ] 4.0
  - [ ] 3.5
  - [ ] 3.4
  - [ ] 3.3

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/SkewJoinOptimizeRule.java [hunk 1]
```diff
@@ -416,6 +441,10 @@ private OptExpression createSkewValueSaltTable(List<ScalarOperator> skewValues,
     private OptExpression addSaltForRightChild(LogicalJoinOperator oldJoinOperator, OptExpression input,
                                                ScalarOperator rightSkewColumn, OptimizerContext context) {
         List<ScalarOperator> skewValues = oldJoinOperator.getSkewValues();
+        // If skew values is empty or contains only NULL, still proceed to create salt table.
```
Comment (m-selmi) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/SkewJoinOptimizeRule.java:444]: would this mean that even if nulls are not skewed, we trigger the salting mechanism (if other values are skewed)?

File: fe/fe-core/src/test/java/com/starrocks/sql/plan/SkewJoinTest.java [hunk 2]
```diff
@@ -324,6 +330,56 @@ public void testIntSkewColumnVarchar() throws Exception {
         assertCContains(sqlPlan, "C_NATIONKEY IN (22, 23, 24, 10, 11)");
     }
 
+    @Test
+    public void testSkewJoinWithNullOnlySkewByStats() throws Exception {
```
Comment (m-selmi) [inline fe/fe-core/src/test/java/com/starrocks/sql/plan/SkewJoinTest.java:334]: We could add a test where both null and some other values are skewed.

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/SkewJoinOptimizeRule.java [hunk 3]
```diff
@@ -167,22 +167,42 @@ public boolean check(OptExpression input, OptimizerContext context) {
...
+        
+        // Check MCV-based skew
         long mcvRowCount = mcvList.stream().mapToLong(pair -> pair.second).sum();
-        return ((double) mcvRowCount / rowCount) > sessionVariable.getSkewJoinDataSkewThreshold();
+        boolean mcvSkew = ((double) mcvRowCount / rowCount) > sessionVariable.getSkewJoinDataSkewThreshold();
+        
+        // Check NULL-based skew - treat high NULL concentration as skew
+        boolean nullSkew = nullsFraction > sessionVariable.getSkewJoinDataSkewThreshold();
+        
+        return mcvSkew || nullSkew;
```
Comment (m-selmi) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/SkewJoinOptimizeRule.java:205]: why are other values handled by summing their occurrence fractions but null is handled separately?

**Review by m-selmi — COMMENTED** [review]
Thanks for creating this :+1:  is there anything missing to move this from draft to open?
